engine/access/state_stream/backend/handler.go

```diff
@@ -215,3 +217,260 @@ func (h *Handler) GetRegisterValues(_ context.Context, request *executiondata.Ge
 
 	return &executiondata.GetRegisterValuesResponse{Values: values}, nil
 }
+
+func (h *Handler) createAccountStatusesFilter(statusFilter *executiondata.StatusFilter) (state_stream.EventFilter, error) {
+	filter := state_stream.EventFilter{}
+	if statusFilter != nil {
+		var err error
+		filter, err = state_stream.NewEventFilter(
+			h.eventFilterConfig,
+			h.chain,
+			statusFilter.GetEventType(),
+			[]string{},
+			[]string{},
+		)
+		if err != nil {
+			return filter, status.Errorf(codes.InvalidArgument, "invalid event filter: %v", err)
+		}
+
+		addresses := statusFilter.GetAddress()
+		if len(addresses) > 0 {
+			var fieldFilters []state_stream.FieldFilter
+			for _, address := range addresses {
+				fieldFilters = append(fieldFilters, state_stream.FieldFilter{
```

Comment:
In the above list, I think we should also maintain the specific `FieldFilter` we need for each type. Generally, it is just `address`, but I think there is at least one exception. Best to be precise